jwql/instrument_monitors/common_monitors/bias_monitor.py

```diff
@@ -0,0 +1,554 @@
+            # Can't insert values with numpy.float32 datatypes into database 
+            # so need to change the datatypes of these values.
+            bias_db_entry = {'aperture': self.aperture,
+                             'uncal_filename': filename,
+                             'cal_filename': processed_file,
+                             'cal_image': output_png,
+                             'expstart': expstart,
+                             'mean': float(mean),
+                             'median': float(median),
+                             'stddev': float(stddev),
+                             'collapsed_rows': collapsed_rows.astype(float),
+                             'collapsed_columns': collapsed_columns.astype(float),
+                             'entry_date': datetime.datetime.now()
+                            }
+            for key in amp_meds.keys():
+                bias_db_entry[key] = float(amp_meds[key])
+            
+            # Add this new entry to the bias database table
+            self.stats_table.__table__.insert().execute(bias_db_entry)
+            logging.info('\tNew entry added to bias database table: {}'
+                         .format(bias_db_entry))
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method.  See module docstrings for further details.
+        """
+
+        logging.info('Begin logging for bias_monitor')
+
+        # Get the output directory and setup a directory to store the data
+        self.output_dir = os.path.join(get_config()['outputs'], 'bias_monitor')
+        ensure_dir_exists(os.path.join(self.output_dir, 'data'))
+
+        # Use the current time as the end time for MAST query
+        self.query_end = Time.now().mjd
+
+        # Loop over all instruments
+        for instrument in ['nircam']:
+            self.instrument = instrument
+
+            # Identify which database tables to use
+            self.identify_tables()
+
+            # Get a list of all possible full-frame apertures for this instrument
+            s = Siaf(self.instrument)
+            possible_apertures = [ap for ap in s.apertures if s[ap].AperType=='FULLSCA']
+
+            for aperture in possible_apertures:
+
+                logging.info('Working on aperture {} in {}'.format(aperture, instrument))
+                self.aperture = aperture
+
+                # Locate the record of the most recent MAST search; use this time
+                # (plus a 30 day buffer to catch any missing files from the previous 
+                # run) as the start time in the new MAST search.
+                most_recent_search = self.most_recent_search()
+                self.query_start = most_recent_search - 30
+
+                # Query MAST for new dark files for this instrument/aperture
+                logging.info('\tQuery times: {} {}'.format(self.query_start, self.query_end))
+                new_entries = mast_query_darks(instrument, aperture, self.query_start, self.query_end)
```

Comment:
Nice use of existing code here.  Perhaps if we use this function more in the future it would be worthwhile pulling it out of the `dark_monitor`, but we can leave it like this for now.

Reply:
yeah, I'm guessing it will be used again - at least for the other NIRCam monitors